src/js.c

```diff
@@ -2656,7 +2656,10 @@ _subscribe(natsSubscription **new_sub, jsCtx *js, const char *subject, const cha
             // set only if jsi->consumer is NULL!
             if (jsi->consumer == NULL)
             {
-                DUP_STRING(s, jsi->consumer, info->Name);
+                if (info->Name != NULL)
+                {
+                    DUP_STRING(s, jsi->consumer, info->Name);
+                }
                 if (s == NATS_OK)
```

Comment:
Not related to this PR, but I wonder if I should not have done the rest of that code for pull consumers only...